Assets/Scripts/Block2TextConsoleManager.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+namespace MoveToCode
+{
+
+    public class Block2TextConsoleManager : Singleton<Block2TextConsoleManager>
+    {
+        StandAloneInstruction curInstruction;
+     
+
+        public void UpdateConsoleOnSnap() //every time a block is snapped, the console is updated with code/text
+        {
+            Block2TextConsoleManager.instance.ClearConsole();
+            curInstruction = StartCodeBlock.instance.GetMyInternalIArgument() as StandAloneInstruction;
+
+            while (curInstruction != null){
+                Block2TextConsoleManager.instance.AddLine(curInstruction?.DescriptiveInstructionToString());
+                curInstruction = curInstruction.GetNextInstruction();
+            }
+
+        }
+
+            /* <------copied from ConsoleManager------>*/
+        TextMeshProUGUI mainConsole, headerConsole;
+        public static string finishedString = "Code Finished";
+
+        void Awake(){
+            ClearConsole();
+        }
+
+        public TextMeshProUGUI GetHeaderConsole() {
+            if (headerConsole == null)
+            {
+                headerConsole = transform.GetChild(0).GetComponent<TextMeshProUGUI>();
+            }
+            return headerConsole;
+        }
+
+        public TextMeshProUGUI GetMainConsole() {
+            if (mainConsole == null)
+            {
+                mainConsole = transform.GetChild(1).GetComponent<TextMeshProUGUI>();
```

Comment:
instead of getchild(1), this should be `GetComponentInChildren<Canvas>` or something with the right component (I know it's my code, I didn't write everything well

Reply:
hmm maybe, but there are two TextMeshProUGUI's -- one in Child 0 (header console) and another in Child 1 (main console)

Reply:
ah ok this is fine for now then